### ISSUE TITLE
hx-boost navigation is blocked by nested in-flight lazy-loaded requests

### ISSUE THREAD

**Issue by kkatpcc** [bigskysoftware/htmx#1441]
When using hx-boost on a typical anchor link driven primary navigation menu, and in the default behavior as a simple `<body>` swapping mechanism, any elements nested within that response `<body>` which contain hx-trigger="load" (lazy loading) causes subsequent use of the original boosted navigation to wait/stall/block until sub-loading is complete.

In other words, if I navigate to hx-boost link 2, which then contains one or more hx-trigger="load" elements in the response, some of which may be very slow loading, then trying to click on hx-boost link 1 or 3 seems to "do nothing" until all other loading is fully complete.

Without hx-boost, clicking on a navigation link abandons all current XHR/AJAX and immediately loads the new link. Based on that stock browser behavior, I would expect hx-boost to work the same without any need for explicit hx-sync coordination magic or any extra configuration really, at least when coupled with hx-get.

**Comment by kkatpcc** [comment]
After some more testing, it looks like the observed blocking behavior is primarily related to server side session locking in my particular development setup case.

After eliminating the server side session locking behavior, hx-boost no longer stalls when clicking on another boosted link.

However, hx-boost still doesn't automatically abort in-flight requests (adding to the original troubleshooting obscurity) which then also causes duplicate in-flight requests if I jump back and forth between pages fast enough. This behavior is technically a separate problem/improvement so I may create a new issue for it after closing this one.
